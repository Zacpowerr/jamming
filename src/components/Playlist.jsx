import TrackList from "./Tracklist";

function Playlist({ trackList, handleSave, removeTrackFromTrackList }) {
  return (
    <div>
      <h1>Playlist</h1>
      <input
        className="input mx-10"
        id="playlist-name"
        type="text"
        defaultValue="New Playlist"
      />
      <TrackList
        trackList={trackList}
        removeTrackFromTrackList={removeTrackFromTrackList}
      />
      <button
        className="button"
        onClick={() => {
          if (document.getElementById("playlist-name").value === "") {
            alert("Please enter a playlist name");
            return;
          } else if (trackList.length === 0) {
            alert("Please add tracks to the playlist");
            return;
          } else {
            handleSave();
          }
        }}
      >
        Save to Spotify
      </button>
    </div>
  );
}
export default Playlist;
