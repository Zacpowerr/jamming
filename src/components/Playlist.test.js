import { render, screen } from "@testing-library/react";
import Playlist from "./Playlist";

describe("Playlist", () => {
  let trackList;
  let handleSave;
  let removeTrackFromTrackList;
  beforeAll(() => {
    trackList = [];
    handleSave = jest.fn();
    removeTrackFromTrackList = jest.fn();
  });

  it("should render correctly", () => {
    const { container } = render(
      <Playlist
        trackList={trackList}
        handleSave={handleSave}
        removeTrackFromTrackList={removeTrackFromTrackList}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render a header", () => {
    render(
      <Playlist
        trackList={trackList}
        handleSave={handleSave}
        removeTrackFromTrackList={removeTrackFromTrackList}
      />
    );
    expect(screen.getByText("Playlist").innerHTML).toBe("Playlist");
  });
  it("should render an input element", () => {
    render(
      <Playlist
        trackList={trackList}
        handleSave={handleSave}
        removeTrackFromTrackList={removeTrackFromTrackList}
      />
    );
    expect(screen.getByDisplayValue("New Playlist")).toBeInTheDocument();
  });
  it("should render button element", () => {
    render(
      <Playlist
        trackList={trackList}
        handleSave={handleSave}
        removeTrackFromTrackList={removeTrackFromTrackList}
      />
    );
    expect(screen.getByRole("button").innerHTML).toBe("Save to Spotify");
  });
});
