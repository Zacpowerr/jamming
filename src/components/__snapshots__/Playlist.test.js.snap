// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist should render correctly 1`] = `
<div>
  <div>
    <h1>
      Playlist
    </h1>
    <input
      class="input mx-10"
      id="playlist-name"
      type="text"
      value="New Playlist"
    />
    <div
      class="tracklist"
    >
      <h2>
        Track List
      </h2>
    </div>
    <button
      class="button"
    >
      Save to Spotify
    </button>
  </div>
</div>
`;
